src/components/search/components/searchMenu/SearchFilters.tsx

```diff
@@ -0,0 +1,204 @@
+
+            /* Set up initial form values from Aggregation */
+            initialValues.filters[aggregationKey] = [];
+
+            if (searchFilters) {
+                searchFilters.forEach((searchFilter) => {
+                    if (Object.keys(aggregations[aggregationKey]).includes(searchFilter)) {
+                        initialValues.filters[aggregationKey].push(searchFilter);
+                    }
+                });
+            }
+
+            /* Add a search query */
+            initialValues[aggregationKey] = '';
+        }
+    });
+
+    return (
+        <Row className="h-100 pb-2">
+            <Col className={`${styles.searchMenu} h-100 overflow-scroll py-3 bg-white`}>
+                {(!isEmpty(aggregations)) &&
+                    <Formik
+                        initialValues={{
+                            ...initialValues
+                        }}
+                        enableReinitialize={true}
+                        onSubmit={async (form) => {
+                            await new Promise((resolve) => setTimeout(resolve, 100));
+
+                            /* Check Free Text Search Query */
+                            if (form.filters.q) {
+                                searchParams.set('q', form.filters.q);
+                            } else {
+                                searchParams.delete('q');
+                            }
+
+                            setSearchParams(searchParams);
+                        }}
+                    >
+                        {({ values, setFieldValue }) => (
+                            <Form>
+                                <Row>
+                                    <Col>
+                                        <p className="c-primary">
+                                            <FontAwesomeIcon icon={faFilter}
+                                                className="px-2"
+                                            />
+                                            Filters
+                                        </p>
+                                    </Col>
+                                    <Col className="col-md-auto">
+                                        <FontAwesomeIcon icon={faChevronLeft}
+                                            className={`${styles.filtersCloseIcon} c-primary`}
+                                            onClick={() => HideFilters()}
+                                        />
+                                    </Col>
+                                </Row>
+                                <Row>
+                                    <Col>
+                                        {Object.keys(filtersList).map((filterKey) => {
+                                            const filter = filtersList[filterKey];
+
+                                            /* Check kind of filter */
+                                            switch (filter.filterType) {
```

Comment:
I don't see mids in this switch is that correct?

Reply:
That's correct, I treated MIDS essentially as a multi select filter, but with a different template. So the MIDS template uses the same functionality as the default Multi Select template. If I would have put MIDS in this select, you would get a lot of repetitive code in the MIDS file that already is in the Multi Select one. That's why I decided to put MIDS a level lower, looking at the type in the Multi Select component to render the default or MIDS template.